3회차/신윤식/4_창용마을무리의개수.py

```diff
@@ -1,3 +1,41 @@
 import sys
 
-sys.stdin = open("_창용마을무리의개수.txt")
+sys.stdin = open("./3회차/신윤식/_창용마을무리의개수.txt")
+T = int(input())
+
+for _ in range(1, T+1):
+    N, M = map(int,input().split())
+    graph = [[] for i in range(N+1)]
+    count = 0
+
+    for j in range(M):
+        a, b = map(int,input().split())
+        graph[a].append(b)
+        graph[b].append(a)
+
+    visited = [0] * (N+1)
+
+    stack = [1]
+    visited[1] = 1
+
+    while True:
+
+        while stack:
+            cur = stack.pop()
+
+            for adj in graph[cur]:
+                if visited[adj] == 0:
+                    visited[adj] = 1
+                    stack.append(adj)
+        
+        count += 1
+
+        if sum(visited) == N:
+            break
+
+        for k in range(1, N+1):
+            if visited[k] == 0:
+                visited[k] = 1
+                stack.append(k)
+                break
+    print(f'#{_} {count}')
```

Comment:
좋은 풀이네요! 👍
DFS로 해결하는 비슷한 일반적인 방법도 소개해드릴게요!
```suggestion
    for i in range(1, n + 1):
        if not visited[i]:
            visited[i] = True
            count += 1
            stack = [i]
            while stack:
                now = stack.pop()
                for adj in graph[now]:
                    if not visited[adj]:
                        visited[adj] = True
                        stack.append(adj)
    print(f'#{_} {count}')
```